pkg/gadgets/profile/tcprtt/tracer/tracer_test.go

```diff
@@ -0,0 +1,695 @@
+				params.Get(ParamExtension).Set("true")
+				return params
+			},
+			expected: expected{
+				config: &Config{
+					remoteAddrHist: true,
+					extension:      true,
+				},
+			},
+		},
+		{
+			description: "by_local_and_remote_address_err",
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamByRemoteAddress).Set("true")
+				params.Get(ParamByLocalAddress).Set("true")
+				return params
+			},
+			expected: expected{
+				err: true,
+			},
+		},
+		{
+			description: "filter_by_local_address",
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamFilterLocalAddress).Set("1.2.3.4")
+				return params
+			},
+			expected: expected{
+				config: &Config{
+					filterLocalAddress: 0x04030201,
+				},
+			},
+		},
+		{
+			description: "filter_by_remote_address",
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamFilterRemoteAddress).Set("192.168.0.1")
+				return params
+			},
+			expected: expected{
+				config: &Config{
+					filterRemoteAddress: 0x0100A8C0,
+				},
+			},
+		},
+		{
+			description: "filter_by_remote_address_and_by_local_address_err",
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamFilterRemoteAddress).Set("192.168.0.1")
+				params.Get(ParamByLocalAddress).Set("true")
+				return params
+			},
+			expected: expected{
+				err: true,
+			},
+		},
+		{
+			description: "filter_by_remote_address_and_by_remote_address_err",
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamFilterRemoteAddress).Set("192.168.0.1")
+				params.Get(ParamByRemoteAddress).Set("true")
+				return params
+			},
+			expected: expected{
+				err: true,
+			},
+		},
+		{
+			description: "filter_by_local_address_and_by_remote_local_address_err",
+			getGadgetParams: func() *params.Params {
+				params := gadget.ParamDescs().ToParams()
+				params.Get(ParamFilterLocalAddress).Set("1.1.1.1")
+				params.Get(ParamByRemoteAddress).Set("true")
+				params.Get(ParamByLocalAddress).Set("true")
+				return params
+			},
+			expected: expected{
+				err: true,
+			},
+		},
+	}
+
+	for _, test := range testTable {
+		test := test
+		t.Run(test.description, func(t *testing.T) {
+			t.Parallel()
+
+			tracer := createTracer(nil)
+			tracer.logger = log.StandardLogger()
+			err := tracer.parseParams(test.getGadgetParams())
+
+			if test.expected.err {
+				require.NotNil(t, err, "expected error parsing params")
+				return
+			} else {
+				require.Nil(t, err, "unexpected error parsing params")
+			}
+
+			require.Equal(t, tracer.config, test.expected.config)
+		})
+	}
+}
+
+func TestRunWithResult(t *testing.T) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+
+	const (
+		serverPort = 8080
+	)
+
+	rand.Seed(time.Now().UTC().UnixNano())
```

Comment:
Mmmmm I think that's true for the integration test because they run as "one unique block". However, for unit-test, I wouldn't know where to create such TestMain as one could run:

```bash
sudo go test -v ./pkg/gadgets/profile/tcprtt/...
```

In that case the seed won't be initialized. Or am I missing something wrt the execution of tests?

Reply:
Good point, it won't work for that case. I think we have two options:
- Create a specific [source](https://pkg.go.dev/math/rand#NewSource) for this test. 
- Don't use random at all and hardcode the IP addresses

Reply:
I used hardcode values for now.